src/v2/api/services/DisciplineTeacherService.spec.ts

```diff
@@ -0,0 +1,207 @@
+import Configuration from '../../config/Configuration';
+import { TelegramConfigService } from '../../config/TelegramConfigService';
+import { NotEnoughAnswersException } from '../../utils/exceptions/NotEnoughAnswersException';
+import { ExcessiveAnswerException } from '../../utils/exceptions/ExcessiveAnswerException';
+import { WrongTimeException } from '../../utils/exceptions/WrongTimeException';
+
+describe('DisciplineTeacherService', () => {
+  let disciplineTeacherService: DisciplineTeacherService;
+  let disciplineRepository: DisciplineRepository;
+  let pollService: PollService;
+  beforeEach(async () => {
+    const testingModule = await Test.createTestingModule({
+      providers: [DisciplineTeacherService, TelegramAPI, TelegramConfigService],
+      imports: [
+        PrismaModule,
+        PollModule,
+        DateModule,
+        MapperModule,
+        ConfigurationModule.forRoot({
+          isGlobal: true,
+          load: [Configuration],
+        })],
+    }).compile();
+
+    disciplineTeacherService = testingModule.get(DisciplineTeacherService);
+    disciplineRepository = testingModule.get(DisciplineRepository);
+    pollService = testingModule.get(PollService);
+  });
+
+  describe('getUniqueQuestions', () => {
+    it('should return array of question', async () => {
+      jest.spyOn(disciplineRepository, 'find').mockImplementation(async () => ({
+        disciplineTeachers: [{
+          id: 'id',
+          roles: [TeacherRole.LECTURER, TeacherRole.PRACTICIAN],
+        }],
+      } as any));
+      jest.spyOn(pollService, 'getQuestions').mockImplementation(() => ([
+        {
+          id: 'question1',
+        },
+        {
+          id: 'question2',
+        }] as any
+      ));
+			
+      const result = await disciplineTeacherService.getUniqueQuestions('id');
+      expect(result).toStrictEqual([
+        {
+          id: 'question1',
+        },
+        {
+          id: 'question2',
+        }]
+      );
+    });
+  });
+
+  describe('checkRequiredQuestions', () => {
+    it('should return undefined if there are enough answers', async () => {
+      const dbQuestions = [{
+        id: 'id',
+        isRequired: true,
+      }] as any;
+      const answers = [{
+        questionId: 'id',
+        value: 5,
+      }] as any;
+
+      const result = await disciplineTeacherService.checkRequiredQuestions(dbQuestions, answers);
+      expect(result).toBeUndefined();
+    });
+
+    it('should trow NotEnoughAnswersException if there are not enough answers', async () => {
+      const dbQuestions = [{
+        id: 'id',
+        isRequired: true,
+      }] as any;
+      const answers = [{
+        questionId: 'id1',
+        value: 5,
+      }] as any;
+      expect.assertions(1);
+      try {
+        await  disciplineTeacherService.checkRequiredQuestions(dbQuestions, answers);
+      } catch (err) {
+        expect(err).toBeInstanceOf(NotEnoughAnswersException);
+      }
```

Comment:
2 spaces after await; since the call is asynchronous it is better to use `.catch((err) => ...)`